rs/rosetta-api/icp_ledger/rosetta-integration-tests/tests/tests.rs

```diff
@@ -1327,3 +1328,142 @@ async fn test_network_status_single_genesis_transaction() {
         genesis_block.block_identifier
     );
 }
+
+#[tokio::test]
+async fn test_rosetta_ledger_upgrade() {
+    let mut env = TestEnv::setup(false, true).await.unwrap();
+
+    // Currently, the ledger canister version is that of this branch
+    // We need to reinstall it to the mainnet version
+    let ledger_wasm_mainnet =
+        std::fs::read(std::env::var("ICP_LEDGER_DEPLOYED_VERSION_WASM_PATH").unwrap()).unwrap();
+    env.pocket_ic
+        .reinstall_canister(
+            env.ledger_id,
+            ledger_wasm_mainnet,
+            icp_ledger_init(env.sender_id),
+            None,
+        )
+        .await
+        .unwrap();
+
+    // Let's create a few transactions to make sure rosetta is working with the mainnet ledger version
+    for _ in 0..100 {
+        let transfer_arg = TransferArg {
+            from_subaccount: None,
+            to: Account::from(Principal::anonymous()),
+            fee: None,
+            created_at_time: None,
+            memo: None,
+            amount: Nat::from(1u64),
+        };
+        let arg = Encode!(&transfer_arg).unwrap();
+        env.pocket_ic
+            .update_call(env.ledger_id, env.sender_id, "icrc1_transfer", arg)
+            .await
+            .unwrap();
+    }
```

Comment:
Would it make sense to here also create send some transactions using Rosetta (same for after upgrading the ledger)?

Reply:
Yes, but we need the rosetta client for this.

Reply:
Are you tracking this in another ticket, or do you already have a draft PR where you do this with the rosetta client?